tests/utils/pvc.go

```diff
@@ -236,3 +236,22 @@ func WaitPVCDeleted(clientSet *kubernetes.Clientset, pvcName, namespace string,
 	})
 	return result, err
 }
+
+// WaitPVCDeletedByUID polls the specified PVC until timeout or it's not found, returns true if the PVC with the same UID is deleted
+// in the specified timeout period, and any errors
+func WaitPVCDeletedByUID(clientSet *kubernetes.Clientset, pvcSpec *k8sv1.PersistentVolumeClaim, timeout time.Duration) (bool, error) {
```

Comment:
The UID doesn't seem important in this fucntion, we are also return true/false, error so I think you can remove the UID part of the function name and description.

Reply:
I beg to differ. The UID is critical in this function. The "regular" `WaitPVCDeleted` is defined in line 224. That function waits for the PVC with the same name to be gone. `WaitForPVCDeletedByUID` waits for deletion of the particular PVC passed as an argument.

Reply:
Okay, so now I completely understand what is going on. and I see the point of the UID. I thought I had asked you to write WaitPVCDeleted, but as you pointed out it already exists above. And this one is discretely different because you have the problem that the CDI controller can (and will) create a new PVC once the original has been deleted, and you need to know that the new PVC is different from the old PVC, so you can conclude the old PVC was actually deleted.